runway/hooks/staticsite/upload_staticsite_env.py

```diff
@@ -0,0 +1,70 @@
+"""CFNgin hook for uploading environment files to a static website S3 bucket."""
+import logging
+import json
+import os
+import yaml
+
+LOGGER = logging.getLogger(__name__)
+
+
+def sync(context, bucket, **kwargs):
+    """Sync static website environment files to S3 bucket.
+
+    Keyword Args:
+
+        context (:class:`runway.cfngin.context.Context`): The context
+            instance.
+        provider (:class:`runway.cfngin.providers.base.BaseProvider`):
+            The provider instance.
+        bucket (str): The static site bucket name.
+        files (List[Dict[str, str]]): List of files and file content that
+            should be uploaded.
+
+    """
+    files = kwargs.get('files', [])
+
+    LOGGER.debug('bucket: %s', bucket)
+    LOGGER.debug('files: %s', json.dumps(files))
+
+    session = context.get_session()
+    s3_client = session.client('s3')
+
+    for env_file in files:
+        filename = env_file.get('name')
+        LOGGER.debug('name: %s', filename)
+
+        if 'content' in env_file:
+            content = env_file.get('content')
+            content_type = 'text/plain'
+
+            _, ext = os.path.splitext(filename)
+            LOGGER.debug('ext: %s', ext)
+
+            if ext == '.json':
+                content = json.dumps(content)
+                content_type = 'application/json'
+            elif ext in ['.yml', '.yaml']:
+                content = yaml.dump(content)
+                content_type = 'text/yaml'
+            else:
+                LOGGER.debug('type: %s', type(content))
+                content = str(content)
```

Comment:
i would almost make content_type need to be explicit, check the type before dumping, and ensure we were able to correctly convert it to a string before continuing. i would also be fine with a combination of autodetecting the content_type and explicitly providing it.

reasoning: there could a file that is json or yaml formatted that does not end with an expected extension. if we skip the last string check and instead just `str()` whatever is there without proper handling of the conversion, it could end up being a useless mess of a file.

```suggestion
        if env_file.get('content'):
            content = env_file['content']
            content_type = env_file.get('content_type', 'text/plain')

            if isinstance(content, (dict, list)):
                if content_type == 'application/json':
                    content = json.dumps(content)
                elif content_type == 'text/yaml':
                    content = yaml.safe_dump(content)
                else:
                    raise ValueError('"content_type" must be provided if '
                                     '"content" is not a string')
            if not isinstance(content, str):
                raise TypeError('unsupported content type: %s' % type(content))
```